src/cloud/validateCode.js

```diff
@@ -60,52 +59,58 @@ const setUserStatus = async (user, reservation = null) => {
     if (reservation) {
       user.set('status', 'inactive');
     } else {
-      // TODO: Uncomment when we use again the currentQuePosition logic.
-      // user.set('quePosition', currentQuePosition);
-      // if (maxQuePosition >= currentQuePosition) {
-      //   user.set('status', 'inactive');
-      // } else {
-      //   user.set('status', 'waitlist');
-      // }
-      user.set('status', 'waitlist');
-      // TODO: Uncomment when the app (frontend) is ready to use it.
-      // await ChatService.createConversation(
-      //   user,
-      //   `${user.id}_invitation_conversation_${new Date().getTime()}`,
-      //   'invitation'
-      // );
-
-      const onboardingRole = await new Parse.Query(Parse.Role)
-        .equalTo('name', ONBOARDING_ADMIN)
-        .first();
-      if (onboardingRole) {
-        // If the role is defined, get the first user with it
+      const hasWaitListConversation = await ChatService.getConversationByCid(
+        `messaging:${user.id}_waitlist_conversation`,
+      );
+
+      if (!hasWaitListConversation) {
+        // TODO: Uncomment when we use again the currentQuePosition logic.
+        // user.set('quePosition', currentQuePosition);
+        // if (maxQuePosition >= currentQuePosition) {
+        //   user.set('status', 'inactive');
+        // } else {
+        //   user.set('status', 'waitlist');
+        // }
+        user.set('status', 'waitlist');
```

Comment:
This probably needs to get set before the if check. They should have a status of waitlist regardless if they have conversation or not.